core/src/forwarding_stage.rs

```diff
@@ -587,51 +580,36 @@ impl TpuClientNextClient {
         bind_socket: UdpSocket,
         cancel: CancellationToken,
     ) -> Self {
-        // For now use large channel, the more suitable size to be found later.
-        let (sender, receiver) = mpsc::channel(128);
         let leader_updater = forward_address_getter.clone();
-
-        let config = Self::create_config(bind_socket, stake_identity);
-        let (update_certificate_sender, update_certificate_receiver) = watch::channel(None);
-        let scheduler: ConnectionWorkersScheduler = ConnectionWorkersScheduler::new(
-            Box::new(leader_updater),
-            receiver,
-            update_certificate_receiver,
-            cancel.clone(),
-        );
-        // leaking handle to this task, as it will run until the cancel signal is received
-        runtime_handle.spawn(scheduler.get_stats().report_to_influxdb(
-            "forwarding-stage-tpu-client",
-            METRICS_REPORTING_INTERVAL,
-            cancel.clone(),
-        ));
-        let _handle = runtime_handle.spawn(scheduler.run(config));
-        Self {
-            sender,
-            update_certificate_sender,
-        }
-    }
-
-    fn create_config(
-        bind_socket: UdpSocket,
-        stake_identity: Option<&Keypair>,
-    ) -> ConnectionWorkersSchedulerConfig {
-        ConnectionWorkersSchedulerConfig {
-            bind: BindTarget::Socket(bind_socket),
-            stake_identity: stake_identity.map(StakeIdentity::new),
+        let (sender, client) = ClientBuilder::with_leader_updater(Box::new(leader_updater))
+            .runtime_handle(runtime_handle.clone())
+            .bind_socket(bind_socket)
+            .identity(stake_identity)
+            .leader_send_fanout(1)
             // Cache size of 128 covers all nodes above the P90 slot count threshold,
             // which together account for ~75% of total slots in the epoch.
-            num_connections: 128,
-            skip_check_transaction_age: true,
-            worker_channel_size: 2,
-            max_reconnect_attempts: 4,
-            // Send to the next leader only, but verify that connections exist
-            // for the leaders of the next `4 * NUM_CONSECUTIVE_SLOTS`.
-            leaders_fanout: Fanout {
-                send: 1,
-                connect: 4,
-            },
-        }
+            .max_cache_size(128)
+            // The channel size represents 8s worth of transactions at a rate of
+            // 1000 tps, assuming batch size is 64.
+            .sender_capacity(128)
+            .worker_channel_size(2)
+            .max_reconnect_attempts(4)
+            .report({
+                |stats: Arc<SendTransactionStats>, cancel: CancellationToken| async move {
+                    stats
+                        .report_to_influxdb(
+                            "forwarding-stage-tpu-client",
+                            METRICS_REPORTING_INTERVAL,
+                            cancel.clone(),
+                        )
+                        .await
```

Comment:
*minor*

You can probably omit explicit argument types.
Also, I'm not sure `cancel` needs to be cloned.

```suggestion
                |stats, cancel| async move {
                    stats
                        .report_to_influxdb(
                            "forwarding-stage-tpu-client",
                            METRICS_REPORTING_INTERVAL,
                            cancel,
                        )
                        .await
```